neuralpp/symbolic/normalizer.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional, Deque
+from collections import deque
+
+from .expression import Expression, Constant, FunctionApplication
+from .z3_expression import Z3SolverExpression
+from .constants import basic_true, basic_false, if_then_else
+from .context_simplifier import ContextSimplifier
+from .parameters import sympy_evaluate
+import neuralpp.symbolic.functions as functions
+
+
+def is_boolean_atom(expression: Expression) -> bool:
+    return expression.type == bool and not isinstance(expression, Constant)  # should ignore `True` and `False`
```

Comment:
We should not split on `True`/`False` as it is meaningless and would cause the algorithm to not terminate. E.g., split on `True` would be `if True then True else False`...

Reply:
:) Thanks for the clarification. It'd be helpful to include your explanation here as comments, so future readers wouldn't be confused